js/admin.js

```diff
@@ -60,23 +70,49 @@
 
 		/**
 		 * Add a new provider
-		 * @returns {number} id of the provider
 		 */
-		addProvider: function(callback) {
-			var providerIds = OCA.User_SAML.Admin.providerIds.split(',');
-			var nextId = 1;
-			if (providerIds.indexOf('1') >= 0) {
-				nextId = 2;
-				while ($.inArray('' + nextId, providerIds) >= 0) {
-					nextId++;
+		addProvider: function (callback) {
+			var xhr = new XMLHttpRequest();
+			xhr.open('PUT', OC.generateUrl('/apps/user_saml/settings/providerSettings'));
+			xhr.setRequestHeader('Content-Type', 'application/json')
+			xhr.setRequestHeader('requesttoken', OC.requestToken)
+
+			xhr.onload = function () {
+				var response = JSON.parse(xhr.response)
+				if (xhr.status >= 200 && xhr.status < 300) {
+					OCA.User_SAML.Admin.providerIds += ',' + response.id;
+					callback(response.id)
+				} else {
+					console.error("Could not fetch new provider ID")
 				}
-			}
-			OCP.AppConfig.setValue('user_saml', 'providerIds', OCA.User_SAML.Admin.providerIds + ',' + nextId, {
-				success: function () {
-					OCA.User_SAML.Admin.providerIds += ',' + nextId;
-					callback(nextId)
+			};
+			xhr.onerror = function () {
+				console.error("Could not fetch new provider ID");
+			};
+
+			xhr.send();
+		},
+
+		updateProvider: function (configKey, configValue, successCb, errorCb) {
+			var xhr = new XMLHttpRequest();
+			xhr.open('POST', OC.generateUrl('/apps/user_saml/settings/providerSettings/' + this.currentConfig));
```

Comment:
somehow it feels like POST and PUT were mixed up

usually in REST we use POST for adding new entries and PUT for updating one